### DISCUSSION TITLE
[Bug]: loadFromJSON not works as expected with custom class (createClass)

### DISCUSSION THREAD

**Discussion by bilalmalkoc** [fabricjs/fabric.js#8551] [Q&A] [answered]
### CheckList

- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I agree to follow this project's Code of Conduct
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.3.0

### In What environments are you experiencing the problem?

Chrome, Safari

### Node Version (if applicable)

None

### Link To Reproduction

https://jsfiddle.net/nxwga37u/

### Steps To Reproduce

1. Open jsFiddle.
2. Click "Add Text" button. It has expected style. (Working good)
3. Click "Add Text (From JSON)" button. It has wrong style.

### Expected Behavior

canvas.loadFromJson() function should track and render createClass modifications like (_renderBackground).

### Actual Behavior

![](https://i.imgur.com/CPILFAS.png)

**Comment by asturur** [comment, marked as answer]
@bilalmalkoc i hope you figured out already, but just in case this is the fixed code:
https://jsfiddle.net/asturur/0cvjg7dn/1/
You missed the 'text' spec on the fromObject function and in the json object the type has to be the same of your class
